### ISSUE TITLE
Benchmarking example configurations reference non-existent class

### ISSUE THREAD

**Issue by ernestum** [HumanCompatibleAI/imitation#664]
## Bug description
In our benchmarking example configurations there are three which reference a `imitation.algorithms.dagger.ExponentialBetaSchedule`, which does not exist in our repository:
- https://github.com/HumanCompatibleAI/imitation/blob/9822af15da30313d5ef86a67a4be0b8fd6ab1109/benchmarking/example_dagger_seals_half_cheetah_best_hp_eval.json#L20
- https://github.com/HumanCompatibleAI/imitation/blob/9822af15da30313d5ef86a67a4be0b8fd6ab1109/benchmarking/example_dagger_seals_walker_best_hp_eval.json#L20
- https://github.com/HumanCompatibleAI/imitation/blob/9822af15da30313d5ef86a67a4be0b8fd6ab1109/benchmarking/example_dagger_seals_hopper_best_hp_eval.json#L20

## Steps to reproduce
For some obscure reason this issue is not caught by CI/CD pipeline. However, when I run the tests locally they fail. 
So to (hopefully) reproduce this, locally checkout `master` and run
```bash
pytest tests/test_benchmarking.py
```
For the same test with more readable output (which reveals which of the config files are the offenders) you can also check out the `fix_benchmark_configs` branch of #665 

**Update:**
Seems like I had outdated package versions. After a `pip install --upgrade --force -e ".[dev]"` the bug was not caught locally as well. Selectively downgrading packages to the previous state revealed that my outdated version of `jsonpickle` was the reason. The bug is detected with `jsonpickle` up to version `2.1.0`. Starting with version `2.2.0` it is ignored.

**So to reproduce do run `pip install jsonpickle==2.1.0` before executing the tests!**

## Environment
- Operating system and version: Linux Mint 5.4.0-135-generic
- Python version: 3.8.10
<details><summary>Output of `pip freeze --all`</summary>
<p>

```
absl-py==1.0.0
aiohttp==3.8.1
aioredis==2.0.1
aiosignal==1.2.0
alabaster==0.7.12
ale-py==0.7.4
alembic==1.7.7
anyio==3.5.0
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
asttokens==2.0.5
async-timeout==4.0.2
attrs==21.4.0
autopage==0.5.0
autopep8==1.6.0
AutoROM==0.4.2
AutoROM.accept-rom-license==0.4.2
Babel==2.10.1
backcall==0.2.0
beautifulsoup4==4.11.1
black==22.6.0
bleach==5.0.0
blessings==1.7
cachetools==5.0.0
certifi==2021.10.8
cffi==1.15.0
cfgv==3.3.1
chai-sacred==0.8.3
chardet==4.0.0
charset-normalizer==2.0.12
click==8.0.4
cliff==3.10.1
cloudpickle==2.1.0
cmaes==0.8.2
cmd2==2.4.1
codecov==2.1.12
codespell==2.1.0
colorama==0.4.4
colorful==0.5.4
colorlog==6.6.0
commonmark==0.9.1
coverage==6.4.4
cycler==0.11.0
Cython==0.29.28
darglint==1.8.1
DataProperty==0.55.0
datasets==2.8.0
debugpy==1.6.0
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.13
dill==0.3.6
distlib==0.3.5
docker-pycreds==0.4.0
docopt==0.6.2
docutils==0.17.1
entrypoints==0.4
exceptiongroup==1.0.0rc8
execnet==1.9.0
executing==0.8.3
fasteners==0.17.3
fastjsonschema==2.15.3
filelock==3.7.1
flake8==4.0.1
flake8-blind-except==0.2.1
flake8-bugbear==22.4.25
flake8-builtins==1.5.3
flake8-commas==2.1.0
flake8-debugger==4.1.2
flake8-docstrings==1.6.0
flake8-isort==4.1.2.post0
fonttools==4.33.3
frozenlist==1.3.0
fsspec==2022.11.0
furo==2022.6.21
gitdb==4.0.9
GitPython==3.1.27
glfw==2.5.3
google==3.0.0
google-api-core==2.7.2
google-auth==2.6.6
google-auth-oauthlib==0.4.6
googleapis-common-protos==1.56.0
gpustat==0.6.0
greenlet==1.1.2
grpcio==1.43.0
gym==0.21.0
huggingface-hub==0.8.1
huggingface-sb3==2.2.3
hypothesis==6.54.4
identify==2.5.5
idna==3.3
imageio==2.18.0
imagesize==1.3.0
-e git+ssh://git@github.com/HumanCompatibleAI/imitation.git@790c03c7d7d7fb43ac0d630974aac0307bcd4f07#egg=imitation
importlab==0.7
importlib-metadata==4.13.0
importlib-resources==5.7.1
iniconfig==1.1.1
ipdb==0.13.9
ipykernel==6.15.2
ipython==8.2.0
ipython-genutils==0.2.0
ipywidgets==7.7.0
isort==5.10.1
jedi==0.18.1
Jinja2==3.1.1
joblib==1.1.0
jsonpickle==1.5.2
jsonschema==4.4.0
jupyter==1.0.0
jupyter-cache==0.5.0
jupyter-client==6.1.12
jupyter-console==6.4.2
jupyter-core==4.10.0
jupyter-server==1.17.0
jupyter-server-mathjax==0.2.5
jupyterlab-pygments==0.2.2
jupyterlab-widgets==1.1.0
kiwisolver==1.4.2
libcst==0.4.7
livereload==2.6.3
lockfile==0.12.2
Mako==1.2.0
Markdown==3.3.6
markdown-it-py==2.1.0
MarkupSafe==2.1.1
matplotlib==3.5.1
matplotlib-inline==0.1.3
mbstrdecoder==1.1.0
mccabe==0.6.1
mdit-py-plugins==0.3.0
mdurl==0.1.2
mistune==0.8.4
msgpack==1.0.3
mujoco-py==2.1.2.14
multidict==6.0.2
multiprocess==0.70.14
munch==2.5.0
mypy==0.982
mypy-extensions==0.4.3
myst-nb==0.16.0
myst-parser==0.18.0
nbclient==0.5.13
nbconvert==6.5.0
nbdime==3.1.1
nbformat==5.3.0
nbsphinx==0.8.9
nest-asyncio==1.5.5
networkx==2.8
ninja==1.10.2.3
nodeenv==1.7.0
notebook==6.4.11
numpy==1.22.3
nvidia-ml-py3==7.352.0
oauthlib==3.2.0
opencensus==0.9.0
opencensus-context==0.1.2
opencv-python==4.5.5.64
optuna==2.10.0
packaging==21.3
pandas==1.4.3
pandoc==2.2
pandocfilters==1.5.0
parso==0.8.3
pathspec==0.9.0
pathtools==0.1.2
pathvalidate==2.5.0
pbr==5.8.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.1.0
pip==22.3.1
pkg_resources==0.0.0
platformdirs==2.5.2
plotly==5.9.0
pluggy==1.0.0
plumbum==1.7.2
ply==3.11
pockets==0.9.1
pre-commit==2.20.0
prettytable==3.2.0
prometheus-client==0.14.1
promise==2.3
prompt-toolkit==3.0.29
protobuf==3.19.4
psutil==5.9.0
psycopg2-binary==2.9.3
ptyprocess==0.7.0
pure-eval==0.2.2
py==1.11.0
py-cpuinfo==8.0.0
py-spy==0.3.11
pyarrow==10.0.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pybullet==3.2.5
pycodestyle==2.8.0
pycparser==2.21
pydocstyle==6.1.1
pydot==1.4.2
pyenchant==3.2.2
pyflakes==2.4.0
pyglet==1.5.27
Pygments==2.12.0
pyparsing==3.0.8
pyperclip==1.8.2
pyrsistent==0.18.1
pytablewriter==0.64.2
pytest==7.1.2
pytest-cov==3.0.0
pytest-env==0.6.2
pytest-forked==1.4.0
pytest-xdist==2.5.0
pytest_notebook==0.8.0
python-dateutil==2.8.2
pytype==2022.7.26
pytz==2022.1
PyYAML==6.0
pyzmq==22.3.0
qtconsole==5.3.0
QtPy==2.0.1
ray==2.0.0
redis==4.3.1
requests==2.27.1
requests-oauthlib==1.3.1
responses==0.18.0
rich==12.6.0
-e git+ssh://git@github.com/HumanCompatibleAI/rl-baselines3-zoo.git@edfd2d707b6347d24bd1b46847c7728ed3f7b554#egg=rl_zoo3
rsa==4.8
sb3-contrib==1.6.2
scikit-learn==1.0.2
scipy==1.9.1
seaborn==0.11.2
seals==0.1.5
Send2Trash==1.8.0
sentry-sdk==1.5.10
setproctitle==1.2.3
setuptools==44.0.0
setuptools-scm==7.0.5
shortuuid==1.0.8
six==1.16.0
smmap==5.0.0
sniffio==1.2.0
snowballstemmer==2.2.0
sortedcontainers==2.4.0
soupsieve==2.3.2.post1
Sphinx==5.1.1
sphinx-autobuild==2021.3.14
sphinx-autodoc-typehints==1.19.2
sphinx-basic-ng==0.0.1a12
sphinx-copybutton==0.5.0
sphinx-github-changelog==1.2.0
sphinx-rtd-theme==1.0.0
sphinx-togglebutton==0.3.2
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-napoleon==0.7
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
sphinxcontrib-spelling==7.5.0
SQLAlchemy==1.4.36
stable-baselines3==1.7.0
stack-data==0.2.0
stevedore==3.5.0
tabledata==1.3.0
tabulate==0.8.10
tcolorpy==0.1.2
tenacity==8.0.1
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorboardX==2.5
terminado==0.13.3
testfixtures==6.18.5
threadpoolctl==3.1.0
tinycss2==1.1.1
tokenize-rt==4.2.1
toml==0.10.2
tomli==2.0.1
torch==1.11.0
tornado==6.1
tqdm==4.64.0
traitlets==5.1.1
typepy==1.3.0
typing-inspect==0.7.1
typing_extensions==4.2.0
urllib3==1.26.9
virtualenv==20.15.1
wandb==0.12.21
wasabi==0.9.1
wcwidth==0.2.5
webencodings==0.5.1
websocket-client==1.3.2
Werkzeug==2.1.1
wheel==0.37.1
widgetsnbextension==3.6.0
wrapt==1.14.0
xxhash==3.2.0
yarl==1.7.2
zipp==3.8.0
```

</p>
</details>

**Comment by ernestum** [comment]
I suspect the issue is the newly introduced `on_missing` parameter in v2.2.0 of jsonpickle:
: https://github.com/jsonpickle/jsonpickle/blob/main/CHANGES.rst#v220
Seems like the default behavior is to just "ignore" non-found classes: 
https://github.com/jsonpickle/jsonpickle/blob/d693ec7dcc2a19ea5e8ec8c076d4ddf04b9f2bab/jsonpickle/unpickler.py#L27

**Comment by ernestum** [comment]
@Rocamonde since you commited the original config files could you give me a hint where to find the missing `imitation.algorithms.dagger.ExponentialBetaSchedule` class? Did it end up in some experimental branch that we deleted?

**Comment by ernestum** [comment]
Turns out the error was not on the `jsonpickle` side. 
We switched the key from `py/object` to `py/type` in #653 to make the `KeyError: 'Invalid key "py/object". Config-keys cannot be one of thereserved jsonpickle tags: [...]` error go away. This obscure error indicated the missing class (which is easy to miss). We could (by chance) make the error go away by swtiching to `py/type` but that obviously did not fix the root cause. This should be fixed in sacred by https://github.com/IDSIA/sacred/pull/902 and we should have better error messages in the future.
